src/main/java/seedu/partyplanet/ui/CommandBox.java

```diff
@@ -36,9 +44,37 @@ public CommandBox(CommandExecutor commandExecutor, AutoCompleter autoCompleter)
         // calls #setStyleToDefault() whenever there is a change to the text of the command box.
         commandTextField.textProperty().addListener((unused1, unused2, unused3) -> setStyleToDefault());
         commandTextField.setOnKeyPressed(e -> handleUserKey(e.getCode()));
+        commandTextField.addEventHandler(KeyEvent.KEY_RELEASED, e -> undoEventHandler(e));
+        commandTextField.addEventHandler(KeyEvent.KEY_RELEASED, e -> redoEventHandler(e));
         history = new InputHistory();
     }
 
+    /**
+     * Handles calling the undo command if CTRL + Z shortcut is used
+     */
+    private void undoEventHandler(KeyEvent event) {
+        if (undo.match(event)) {
+            try {
+                commandExecutor.execute("undo");
+            } catch (CommandException | ParseException e) {
+                setStyleToIndicateCommandFailure();
+            }
+        }
+    }
+
+    /**
+     * Handles calling the redo command if CTRL + SHIFT + Z or CTRL + R shortcut is used
+     */
+    private void redoEventHandler(KeyEvent event) {
+        if (redo1.match(event) || redo2.match(event)) {
+            try {
+                commandExecutor.execute("redo");
+            } catch (CommandException | ParseException e) {
+                setStyleToIndicateCommandFailure();
+            }
+        }
+    }
```

Comment:
Suggestion to use `setOnKeyReleased` then no need use a generic event handler.
Also to combine duplicated event handlers

```suggestion
        commandTextField.setOnKeyReleased(e -> handleUndoRedo(e));
        history = new InputHistory();
    }

    /**
     * Handles key combination releases.
     * Exceutes undo command if CTRL + Z shortcut is used
     * Exceutes redo command if CTRL + SHIFT + Z or CTRL + R shortcut is used
     */
    private void handleUndoRedo(KeyEvent event) {
        if (undo.match(event)) {
            try {
                commandExecutor.execute("undo");
            } catch (CommandException | ParseException e) {
                setStyleToIndicateCommandFailure();
            }
        } else if (redo1.match(event) || redo2.match(event)) {
            try {
                commandExecutor.execute("redo");
            } catch (CommandException | ParseException e) {
                setStyleToIndicateCommandFailure();
            }
        }
    }
```

Reply:
Incorporated into latest commit, thanks!